hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/NonBlockingIOThread.java

```diff
@@ -171,10 +190,15 @@ public final void run() {
         try {
             for (; ; ) {
                 try {
-                    if (selectNow) {
-                        runSelectNowLoop();
-                    } else {
-                        runSelectLoop();
+                    switch (selectorMode) {
+                        case CPU_WORKAROUND:
+                            runSelectLoopWithSelectorFix();
+                            break;
+                        case NOW:
+                            runSelectNowLoop();
+                            break;
+                        default:
```

Comment:
I would not use 'default' for runSelect loop. Add a third 'case' and in the default an exception can be thrown.